src/Repositories/BlockRepository.php

```diff
@@ -35,42 +33,37 @@ public function getCrops(string $role): array
 
     public function hydrate(TwillModelContract $model, array $fields): TwillModelContract
     {
-        if (Schema::hasTable(config('twill.related_table', 'twill_related'))) {
-            $relatedItems = Collection::make();
-
-            Collection::make($fields['browsers'])->each(function ($items, $browserName) use (&$relatedItems) {
-                Collection::make($items)->each(function ($item) use ($browserName, &$relatedItems) {
-                    try {
-                        // @todo: Repository could be null.
-                        $repository = $this->getModelRepository($item['endpointType'] ?? $browserName);
-                        $relatedItems->push(
-                            (object) [
-                                'related' => $repository->getById($item['id']),
-                                'browser_name' => $browserName,
-                            ]
-                        );
-                    } catch (ReflectionException $reflectionException) {
-                        Log::error($reflectionException);
-                    }
-                });
-            });
-
-            $model->setRelation('relatedItems', $relatedItems);
-        }
+        $relatedItems = collect($fields['browsers'])
+            ->flatMap(fn($items, $browserName) => collect($items)
+                ->map(fn($item, $position) => new RelatedItem([
+                    'subject_id' => $model->getKey(),
+                    'subject_type' => $model->getMorphClass(),
+                    'related_id' => $item['id'],
+                    'related_type' => $item['endpointType'],
+                    'browser_name' => $browserName,
+                    'position' => $position,
+                ])
+            )
+        );
+
+        $model->setRelation('relatedItems', $relatedItems);
```

Comment:
Maybe here if we're gonna use the items anyways we can do $model->loadMissing('relatedItems.related') to do just one eager load query instead of one query per model